packages/nimble-components/src/wafer-map/modules/experimental/prerendering.ts

```diff
@@ -0,0 +1,86 @@
+import { scaleLinear } from 'd3-scale';
+import { ticks } from 'd3-array';
+import { WaferMapColorScaleMode } from '../../types';
+import type { ColorScale, Dimensions } from '../../types';
+import type { WaferMap } from '../..';
+
+/**
+ * Prerendering prepares render-ready dies data to be used by the rendering module
+ */
+export class Prerendering {
+    public get labelsFontSize(): number {
+        return this._labelsFontSize;
+    }
+
+    public get colorScale(): ColorScale {
+        return this._colorScale;
+    }
+
+    private _colorScale!: ColorScale;
+
+    private _labelsFontSize!: number;
+
+    private readonly fontSizeFactor = 0.8;
+    private readonly colorScaleResolution = 10;
+
+    public constructor(private readonly wafermap: WaferMap) {}
+
+    public update(): void {
```

Comment:
This seems, to me, an awkward dependency pattern. It seems strange to have an `update` method that the call site actually has no semantic idea of what that `update` does. Calling `update` on the pre-renderer for every various `update<x>` method in the `DataManager` doesn't seem like an obvious requirement. Is it? How do know when to use/not use it?

The fact that there is nothing observable happening with respect to the changes to the `_colorScale` and `_labelsFontSize` properties is also concerning.

While I don't have a precise suggestion for a better direction to move in, I would recommend looking at other patterns we have that have _observable_ dependencies from one to the other, like the `ColumnInternals` or the `Virtualizer`.

Reply:
Sure! I created this Issue to track these changes after we have all the pieces of the new update together https://github.com/ni/nimble/issues/1984